### PR TITLE
Optimize log_validation in train_controlnet_flax

### PR DESCRIPTION
**Pull request by cgarciae**
# Changes
* Extracts `pipeline` from `log_validation` to avoid recompilation every time the function is called.

### CODE REVIEW

File: examples/controlnet/train_controlnet_flax.py [hunk 1]
```diff
@@ -76,20 +76,11 @@ def image_grid(imgs, rows, cols):
...
-        tokenizer=tokenizer,
-        controlnet=controlnet,
-        safety_checker=None,
-        dtype=weight_dtype,
-        revision=args.revision,
-        from_pt=args.from_pt,
-    )
-    params = jax_utils.replicate(params)
-    params["controlnet"] = controlnet_params
+    pipeline_params = pipeline_params.copy()
```
Comment (pcuenca) [inline examples/controlnet/train_controlnet_flax.py:82]: Why do we need to copy the params?

File: examples/controlnet/train_controlnet_flax.py [hunk 2]
```diff
@@ -176,6 +167,7 @@ def save_model_card(repo_id: str, image_logs=None, base_model=str, repo_folder=N
 - text-to-image
 - diffusers
 - controlnet
+- jax-diffusers-event
```
Comment (pcuenca) [inline examples/controlnet/train_controlnet_flax.py:170]: Cool, we just need to remember this is specific for the event and remove later.

**Review by pcuenca — APPROVED** [review]
Thanks, this should help speed things up!

**Comment by pcuenca** [comment]
Can you please try to run `make style` to make the linter happy? :)

File: examples/controlnet/train_controlnet_flax.py (hunk 2 above)
Comment (yiyixuxu) [reply examples/controlnet/train_controlnet_flax.py:170]:
```suggestion
```
think we can remove this line here and keep it in the script on the event repo

**Review by yiyixuxu — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
Cool!

**Pull request merged**